src/commands/guildLog.ts

```diff
@@ -0,0 +1,189 @@
+								counter++;
+							}
+							if (counter == 10) {
+								bot.logger.log("Counter is 10!");
+								bot.logger.log("Matches found!\n" + logEntry);
+								bot.mineflayer.removeListener("message", chatListener);
+
+								const embed = new EmbedBuilder()
+									.setAuthor({
+										name: "Guild Log",
+										iconURL:
+											"https://media.discordapp.net/attachments/522930879413092388/849317688517853294/misc.png",
+									})
+									.setColor("Red")
+									.setFooter({ text: `Made with <3 from Vliegenier04` })
+									.setThumbnail(
+										`https://media.discordapp.net/attachments/522930879413092388/849317688517853294/misc.png`,
+									)
+									.setTimestamp()
+									.setTitle(`Guild Logger`)
+									.setURL(`https://vliegenier04.dev`)
+									.setDescription(`This is Guild Log\n\n ${logEntry}`);
+
+								await interaction.reply({ embeds: [embed] });
+							}
+						}
+					}),
+				);
+			} catch (err) {
+				bot.logger.error(err);
+			}
+		} else if (member == undefined) {
+			try {
+				await bot.executeTask(`/g log ${page}`);
+				const gLogArray: string[] = [];
+				let chatListener: BotEvents["message"];
+				const regex =
+					/(([A-Za-z]{3}\s[0-9]{1,2}\s[0-9]{4})\s(([0-9]{2}):([0-9]{2}))\s((EDT|EST): ))([A-Za-z-0-9-_]{2,27})\s(joined|left|invited|kicked)(?:\s)?([A-Za-z-0-9-_]{2,27})?(?:\s)?([A-Za-z-0-9-\/\.!-_-\s]+)?/;
+				bot.mineflayer.on(
+					"message",
+					(chatListener = async (message) => {
+						bot.logger.error("Trying to match regex...");
+						const messageContent = message.toString();
+						const match = regex.exec(messageContent);
+
+						let counter = 1;
+
+						while (match && counter < 10) {
+							const logEntry = match[0];
+							if (logEntry) {
+								const logEntryLines = logEntry.split("\n");
+								const filteredLines = logEntryLines.filter((line) => regex.test(line));
+								gLogArray.push(filteredLines.join("\n"));
+								counter++;
+							}
+							if (counter == 10) {
+								bot.logger.log("Counter is 10!");
+								bot.logger.log("Matches found!\n" + logEntry);
+								bot.mineflayer.removeListener("message", chatListener);
+
+								const embed = new EmbedBuilder()
+									.setAuthor({
+										name: "Guild Log",
+										iconURL:
+											"https://media.discordapp.net/attachments/522930879413092388/849317688517853294/misc.png",
+									})
+									.setColor("Red")
+									.setFooter({ text: `Made with <3 from Vliegenier04` })
+									.setThumbnail(
+										`https://media.discordapp.net/attachments/522930879413092388/849317688517853294/misc.png`,
+									)
+									.setTimestamp()
+									.setTitle(`Guild Logger`)
+									.setURL(`https://vliegenier04.dev`)
+									.setDescription(`This is Guild Log ${page}\n\n ${logEntry}`);
+
+								await interaction.reply({ embeds: [embed] });
+							}
+						}
+					}),
+				);
+			} catch (err) {
+				bot.logger.error(err);
+			}
+		} else {
+			try {
```

Comment:
Haven't looked through it super thoroughly, but is this just a copy paste of the above?

Reply:
Yes, required for each block.

Reply:
Ahh... it's not. But a large part of the command could be added to a function so you aren't repeating yourself